locations/paas/src/main/java/brooklyn/location/PaasProvisioningLocation.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package brooklyn.location;
+
+public interface PaasProvisioningLocation<T extends PaasLocation> extends ProvisioningLocation<T> {
```

Comment:
Our location may contain a CF client instance which was set up. Probably, we may define a new hierarchy to express that PaaSLocation could be setup (using the User credentials, end_point, etc). So, the PaaSLocation could provide a methods to instance (and setup) a CF client, which will be maintained by the PaaSLocation (similar to JCloudsLocation that maintains a set of JCloudsSshMachineLocation).

WDYT?